code/game/machinery/transformer.dm

```diff
@@ -35,163 +61,104 @@
-			AM.loc = src.loc
-			do_transform(AM)
+	var/mob/living/carbon/human/H = AM
+	var/move_dir = get_dir(loc, H.loc)
 
-/obj/machinery/transformer/proc/do_transform(var/mob/living/carbon/human/H)
-	if(stat & (BROKEN|NOPOWER))
-		return
-	if(cooldown == 1)
+	if((transform_standing || H.lying) && move_dir == acceptdir)
+		H.forceMove(drop_location())
+		do_transform(H)
+
+/// Transforms a human mob into a cyborg, connects them to the malf AI which placed the factory.
+/obj/machinery/transformer/proc/do_transform(mob/living/carbon/human/H)
+	if(is_on_cooldown || stat & (BROKEN|NOPOWER))
 		return
 
 	if(!transform_dead && H.stat == DEAD)
-		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
+		playsound(loc, 'sound/machines/buzz-sigh.ogg', 50, 0)
 		return
 
-	playsound(src.loc, 'sound/items/welder.ogg', 50, 1)
-	H.emote("scream") // It is painful
-	H.adjustBruteLoss(max(0, 80 - H.getBruteLoss())) // Hurt the human, don't try to kill them though.
-
-	// Sleep for a couple of ticks to allow the human to see the pain
-	sleep(5)
-
+	playsound(loc, 'sound/items/welder.ogg', 50, 1)
 	use_power(5000) // Use a lot of power.
-	var/mob/living/silicon/robot/R = H.Robotize(1) // Delete the items or they'll all pile up in a single tile and lag
-
-	R.cell.maxcharge = robot_cell_charge
-	R.cell.charge = robot_cell_charge
-
- 	// So he can't jump out the gate right away.
-	R.lockcharge = !R.lockcharge
-	spawn(50)
-		playsound(src.loc, 'sound/machines/ping.ogg', 50, 0)
-		sleep(30)
-		if(R)
-			R.lockcharge = !R.lockcharge
-			R.notify_ai(1)
 
 	// Activate the cooldown
-	cooldown = 1
+	is_on_cooldown = TRUE
 	update_icon()
-	spawn(cooldown_duration)
-		cooldown = 0
-		update_icon()
-
-/obj/machinery/transformer/conveyor/New()
-	..()
-	var/turf/T = loc
-	if(T)
-		// Spawn Conveyour Belts
-
-		//East
-		var/turf/east = locate(T.x + 1, T.y, T.z)
-		if(istype(east, /turf/simulated/floor))
-			new /obj/machinery/conveyor/auto(east, WEST)
-
-		// West
-		var/turf/west = locate(T.x - 1, T.y, T.z)
-		if(istype(west, /turf/simulated/floor))
-			new /obj/machinery/conveyor/auto(west, WEST)
+	addtimer(CALLBACK(src, .proc/reset_cooldown), cooldown_duration)
+	addtimer(CALLBACK(null, .proc/playsound, loc, 'sound/machines/ping.ogg', 50, 0), 3 SECONDS)
 
+	H.emote("scream")
+	if(!masterAI) // If the factory was placed via admin spawning or other means, it wont have an owner_AI.
+		H.Robotize(robot_cell_type)
+		return
 
+	var/mob/living/silicon/robot/R = H.Robotize(robot_cell_type, FALSE, masterAI)
+	if(R.mind && !R.client && !R.grab_ghost()) // Make sure this is an actual player first and not just a humanized monkey or something.
+		message_admins("[key_name_admin(R)] was just transformed by a borg factory, but they were SSD. Polling ghosts for a replacement.")
+		var/list/candidates = pollCandidates("Do you want to play as a malfunctioning cyborg?", ROLE_TRAITOR, poll_time = 15 SECONDS)
+		if(!LAZYLEN(candidates))
```

Comment:
just `length` here will do. `LAZYLEN` will now only be used for lists that are initialised lazily.

Reply:
Can do.